dev/io.openliberty.cdi.4.1.internal_fat/fat/src/io/openliberty/cdi41/internal/fat/invokers/app/InvokersTestServlet.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.cdi41.internal.fat.invokers.app;
+
+import static io.openliberty.cdi41.internal.fat.invokers.app.ExceptionAssertions.assertThrows;
+import static io.openliberty.cdi41.internal.fat.invokers.app.ExceptionAssertions.exception;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import jakarta.inject.Inject;
+import jakarta.servlet.annotation.WebServlet;
+
+@SuppressWarnings("serial")
+@WebServlet("/invokersTest")
+public class InvokersTestServlet extends FATServlet {
+
+    @Inject
+    private InvokersExtension ext;
+
+    @Inject
+    private InvokedBean bean;
+
+    @Test
+    public void testNullInstance() {
+        // Cannot invoke {0} because the instance passed to the Invoker was null
+        assertThrows(() -> {
+            ext.getBasic().invoke(null, new Object[] { 1 });
+        }, exception().ofType(NullPointerException.class)
+                      .messageIncludes("Cannot invoke")
+                      .messageIncludes("basicMethod")
+                      .messageIncludes("the instance passed to the Invoker was null"));
+    }
+
+    @Test
+    public void testWrongInstanceType() {
+        // Cannot invoke {0} because the instance passed to the Invoker has type {1} which cannot be cast to {2}
+        assertThrows(() -> {
+            ext.getBasic().invoke(new Object(), new Object[] { 1 });
+        }, exception().ofType(ClassCastException.class)
+                      .messageIncludes("Cannot invoke")
+                      .messageIncludes("basicMethod")
+                      .messageIncludes("java.lang.Object")
+                      .messageIncludes("cannot be cast")
+                      .messageIncludes("InvokedBean"));
+    }
+
+    @Test
+    public void testMissingArguments() {
+        // Cannot invoke {0} because {1} arguments were expected but only {2} were provided
+        assertThrows(() -> {
+            ext.getBasic().invoke(bean, new Object[] {});
+        }, exception().ofType(IllegalArgumentException.class)
+                      .messageIncludes("Cannot invoke")
+                      .messageIncludes("basicMethod")
+                      .messageIncludes("1 arguments were expected but only 0 were provided"));
+    }
+
+    @Test
+    public void testNullArgumentsArrayWhenArgsRequired() {
+        // Cannot invoke {0} because the args parameter is null and arguments are required
+        assertThrows(() -> {
+            ext.getBasic().invoke(bean, null);
+        }, exception().ofType(NullPointerException.class)
+                      .messageIncludes("Cannot invoke")
+                      .messageIncludes("basicMethod")
+                      .messageIncludes("the args parameter is null and arguments are required"));
+    }
+
+    @Test
+    public void testNullArgsArrayWithNoargs() throws Exception {
+        ext.getNoargs().invoke(bean, null);
```

Comment:
I assume this case is not expected to throw an Exception and that we are treating `null` as akin to an emptyArray

a comment in similar fashion to the others would be helpful

Reply:
Added a comment to this test. I realised that the others all have the expected exception message pattern as the comment which is why I'd missed it on the case that doesn't throw an exception.

Also added a few more cases that I hadn't thought of.